tests/visualization_tests/test_contour.py

```diff
@@ -159,6 +159,7 @@ def test_plot_contour(
         figure.write_image(BytesIO())
     else:
         plt.savefig(BytesIO())
+        plt.close()
```

Comment:
Without this, 

```
tests/visualization_tests/test_contour.py ...............................................................................                                                               [100%]

====================================================================================== warnings summary =======================================================================================
tests/visualization_tests/test_contour.py::test_plot_contour[_create_study_with_log_scale_and_str_category_3d-None-plot_contour1]
  /opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/optuna-3.1.0.dev0-py3.9.egg/optuna/visualization/matplotlib/_contour.py:132: RuntimeWarning:

  More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`).
```

Reply:
We might run into this problem again if we add more tests where figure needs to be opened. I think approach in this PR is fine for now, as most of tests actually check `_ContourInfo` object, but maybe it's worth considering adding at some point a small context manager wrapping both `plot_contour` functions and handling setup and tear down.